### ISSUE TITLE
Mind Control bugs

### ISSUE THREAD

**Issue by ajakaja** [dalaranwow/dalaran-wow#5366]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: I've been leveling a priest and just soloed the SM graveyard and library at around level 34-36, which would not be possible, except for the many bugs that made it easier than it should have been.

1) When Mind controlling a mob, when combat ends the mob will glitch out -- MC will end and it will reset, but no longer respond to attacks and become freely killable.
2) MC'd monsters continue to follow their AI scripts. For example, a patrolling mob will continue patrolling, and will only attack when I give an auto-attack command.
3) Mind control does not break when my character is attacked.
4) Mind control cannot be canceled by clicking 'cancel' on the action bar (and I'm pretty sure in live there was a buff you could click off, also, but not 100% sure)
5) Some mobs' abilities have no cooldowns when mind-controlled. For example, I MC'd the Scarlet Monks in the Scarlet Monastery Graveyard and was able to kill other mobs in 10s by spamming their flurry+kick abilities several times per second.

**Steps to reproduce the problem**:

Should be easy to reproduce by going to SM and mind-controlling a Scarlet Monk and killing whatever other mobs are around with it.

**Include proofs for this behaviour**

Sorry, not sure how, but it's very obviously not working correctly since it should not be possible to glitch out a mob and then kill it without repercussion, which trivializes soloing some instances.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

It is easiest to replicate all of these with Scarlet Monks, which are NPC #51659, though I believe it happens with all mind-controllable mobs (at least, I've seen it in many other places also.) 
Scarlet Monk: http://www.wowhead.com/npc=51659/scarlet-monk

**Comment by ajakaja** [comment]
I realize now that there's another issue for these that didn't come up when I searched for some reason: https://github.com/dalaranwow/dalaran-wow/issues/3314
